truffle/src/main/java/org/jruby/truffle/nodes/core/EncodingNodes.java

```diff
@@ -217,6 +314,21 @@ public RubyEncoding defaultExternal(RubyEncoding encoding) {
             return encoding;
         }
 
+        @Specialization
+        public RubyEncoding defaultExternal(RubyString encodingString) {
+            notDesignedForCompilation();
+
+            final RubyEncoding rubyEncoding = RubyEncoding.getEncoding(encodingString.toString());
+            getContext().getRuntime().setDefaultExternalEncoding(rubyEncoding.getEncoding());
```

Comment:
This sets the default external encoding on the JRuby runtime. I'm not sure if we re-use that same value in Truffle. But since it passes the specs I won't worry too much.